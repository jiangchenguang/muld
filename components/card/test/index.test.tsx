import * as React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Card from '..';

describe('Card', () => {
    afterEach(cleanup);
    it('click event', () => {
        const props = {
            onClick: jest.fn(),
        };
        const { container } = render(<Card {...props}>content</Card>);
        const card = container.firstChild;
        expect(card).toHaveClass('mul-card');
        fireEvent.click(card!);
        expect(props.onClick).toHaveBeenCalled();
    });

    it('click-thumb event', () => {
        const onClickThumb = jest.fn();
        const { container } = render(
            <Card thumb="x" onClickThumb={onClickThumb}>
                content
            </Card>,
        );
        const thumb = container.querySelector('.mul-card__thumb')!;
        fireEvent.click(thumb);
        expect(onClickThumb).toHaveBeenCalled();
    });

    it('render price & num', () => {
        const { asFragment } = render(
            <Card num="Custom Num" price="Custom Price">
                content
            </Card>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('render origin-price', () => {
        const { asFragment } = render(<Card originPrice="Custom Origin Price">content</Card>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('render bottom', () => {
        const { asFragment } = render(<Card bottom={'Custom Bottom'}>content</Card>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('render thumb & tag', () => {
        const { asFragment } = render(
            <Card tag="Custom Tag" thumb="Custom Thumb">
                content
            </Card>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('render title & desc', () => {
        const { asFragment } = render(
            <Card title="Custom Title" desc="Custom desc">
                content
            </Card>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('render price & priceTop', () => {
        const { asFragment } = render(
            <Card price="Custom Price" priceTop="Custom Price-top">
                content
            </Card>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
