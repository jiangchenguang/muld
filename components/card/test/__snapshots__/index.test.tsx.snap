// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card render bottom 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm hlOa-dQ mul-card"
  >
    <div
      class="mul-card__header"
    >
      <div
        class="mul-card__content"
      >
        <div />
        <div
          class="mul-card__bottom"
        >
          Custom Bottom
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card render origin-price 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm hlOa-dQ mul-card"
  >
    <div
      class="mul-card__header"
    >
      <div
        class="mul-card__content"
      >
        <div />
        <div
          class="mul-card__bottom"
        >
          <div
            class="mul-card__origin-price"
          >
            ¥ Custom Origin Price
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card render price & num 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm hlOa-dQ mul-card"
  >
    <div
      class="mul-card__header"
    >
      <div
        class="mul-card__content"
      >
        <div />
        <div
          class="mul-card__bottom"
        >
          <div
            class="mul-card__price"
          >
            <div>
              <span
                class="mul-card__price-currency"
              >
                ¥
              </span>
              <span
                class="mul-card__price-integer"
              >
                Custom Price
              </span>
              .
              <span
                class="mul-card__price-decimal"
              />
            </div>
          </div>
          <div
            class="mul-card__num"
          >
            xCustom Num
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card render price & priceTop 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm hlOa-dQ mul-card"
  >
    <div
      class="mul-card__header"
    >
      <div
        class="mul-card__content"
      >
        <div />
        <div
          class="mul-card__bottom"
        >
          Custom Price-top
          <div
            class="mul-card__price"
          >
            <div>
              <span
                class="mul-card__price-currency"
              >
                ¥
              </span>
              <span
                class="mul-card__price-integer"
              >
                Custom Price
              </span>
              .
              <span
                class="mul-card__price-decimal"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card render thumb & tag 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm hlOa-dQ mul-card"
  >
    <div
      class="mul-card__header"
    >
      <a
        class="mul-card__thumb"
      >
        <div
          class="sc-fzpans edAHYF mul-image"
          style="width: 100%; height: 100%;"
        >
          <img
            class="mul-image__img"
            src="Custom Thumb"
            style="object-fit: cover;"
          />
          <div
            class="mul-image__loading"
          >
            <i
              class="sc-fznZeY gWPLVv mul-image__loading-icon mul-image__loading-icon-photo"
              content="F09D"
              name="photo"
              style="font-size: 20px;"
            />
          </div>
        </div>
        <div
          class="mul-card__tag"
        >
          <span
            class="sc-AxirZ dGXMAt mul-tag mul-tag--mark mul-tag--danger"
          >
            Custom Tag
          </span>
        </div>
      </a>
      <div
        class="mul-card__content"
      >
        <div />
        <div
          class="mul-card__bottom"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card render title & desc 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm hlOa-dQ mul-card"
  >
    <div
      class="mul-card__header"
    >
      <div
        class="mul-card__content"
      >
        <div>
          <div
            class="mul-card__title mul-multi-ellipsis--l2"
          >
            Custom Title
          </div>
          <div
            class="mul-card__desc mul-ellipsis"
          >
            Custom desc
          </div>
        </div>
        <div
          class="mul-card__bottom"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
